### PR TITLE
Feature/add remove custom taste

### PR DESCRIPTION
**Pull request by aLLeXUs**

### CODE REVIEW

File: frontend/src/components/taste/TasteList.vue [hunk 1]
```diff
@@ -73,10 +94,17 @@ export default {
...
         ...mapState('taste', ['allTastes', 'myTastes']),
-        ...mapGetters('taste', ['getMyTastesIds'])
+        ...mapGetters('taste', ['getMyTastesIds', 'getAllTastesIds', 'getTasteByName'])
     },
     created() {
-        this.$store.dispatch('taste/fetchTastes');
+        this.$store.dispatch('taste/fetchTastes').then(() => {
+            let customTastes = [];
+            this.getAllTastesIds.forEach((id) => {
+                if (this.allTastes.byId[id].user_id)
```
Comment (nemoipaha) [inline frontend/src/components/taste/TasteList.vue]: this.allTastes.byId[id].user_id should be equal to current user's id?
always use if (this.allTastes.byId[id].user_id) `{...}`

File: frontend/src/store/modules/taste/actions.js [hunk 2]
```diff
@@ -49,5 +49,31 @@ export default {
...
         });
+    },
+    addCustomTaste: (context, name) => {
+        return new Promise((resolve, reject) => {
+            httpService.post('/tastes', {
+                name: name
+            })
+                .then(function (res) {
+                    let taste = res.data.data;
+                    context.commit('ADD_TASTE', taste);
```
Comment (nemoipaha) [inline frontend/src/store/modules/taste/actions.js]: ADD_CUSTOM_TASTE is better

File: frontend/src/store/modules/taste/actions.js [hunk 3]
```diff
@@ -49,5 +49,31 @@ export default {
                     reject(err);
                 });
         });
+    },
+    addCustomTaste: (context, name) => {
+        return new Promise((resolve, reject) => {
+            httpService.post('/tastes', {
```
Comment (nemoipaha) [inline frontend/src/store/modules/taste/actions.js]: /tastes/custom more clear

File: frontend/src/store/modules/taste/actions.js [hunk 4]
```diff
@@ -49,5 +49,31 @@ export default {
...
+                }).catch(function (err) {
+                    reject(err);
+                });
+        });
+    },
+    deleteCustomTaste: (context, id) => {
+        return new Promise((resolve, reject) => {
+            httpService.delete('/tastes/' + id)
+                .then(function (res) {
+                    context.commit('DELETE_TASTE', id);
```
Comment (nemoipaha) [inline frontend/src/store/modules/taste/actions.js]: DELETE_CUSTOM_TASTE

File: frontend/src/store/modules/taste/actions.js [hunk 5]
```diff
@@ -49,5 +49,31 @@ export default {
...
+                    context.commit('ADD_TASTE', taste);
+                    resolve(res);
+                }).catch(function (err) {
+                    reject(err);
+                });
+        });
+    },
+    deleteCustomTaste: (context, id) => {
+        return new Promise((resolve, reject) => {
+            httpService.delete('/tastes/' + id)
```
Comment (nemoipaha) [inline frontend/src/store/modules/taste/actions.js]: /tastes/custom is better

File: app/Http/Controllers/Api/User/UserTaste/TasteController.php [hunk 6]
```diff
@@ -2,21 +2,56 @@
...
+            $addTasteResponse = $this->addTasteAction->execute(
+                new AddTasteRequest($request['name'])
+            );
+            return $this->successResponse($addTasteResponse->getTaste(), 201);
+        } catch (\Exception $exception) {
+            return $this->errorResponse($exception->getMessage(), 400);
+        }
+    }
+
+    public function deleteTaste(int $tasteId)
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/User/UserTaste/TasteController.php]: User can delete only own taste, so validation is required here

File: frontend/src/store/modules/taste/getters.js [hunk 7]
```diff
@@ -5,5 +5,23 @@ export default {
...
+        if (state.allTastes.byId !== undefined) {
+            return Object.keys(state.allTastes.byId);
+        } else {
+            return [];
+        }
+    },
+    getTasteByName: (state, getters) => name => {
+        let ids = getters.getAllTastesIds;
+        let taste = null;
+        ids.forEach(id => {
```
Comment (nemoipaha) [inline frontend/src/store/modules/taste/getters.js]: a bit shorter )
const id = ids.find(id => {
            return state.allTastes.byId[id].name === name;
});

return id ? state.allTastes.byId[id] : null;

File: routes/api.php [hunk 8]
```diff
@@ -98,6 +98,10 @@
         Route::prefix('tastes')->group(function () {
             Route::get('/', 'Api\User\UserTaste\TasteController@getTastes')
                 ->name('tastes.getTastes');
+            Route::post('/', 'Api\User\UserTaste\TasteController@addTaste')
```
Comment (nemoipaha) [inline routes/api.php]: /custom route

File: routes/api.php [hunk 9]
```diff
@@ -98,6 +98,10 @@
...
+                ->name('tastes.addTaste');
+            Route::delete('/{id}', 'Api\User\UserTaste\TasteController@deleteTaste')
```
Comment (nemoipaha) [inline routes/api.php]: same

File: tests/Feature/Api/UserTaste/TasteTest.php [hunk 10]
```diff
@@ -37,4 +37,31 @@ public function test_get_tastes()
                     ->assertJsonCount(3, 'data')
                     ->assertOk();
     }
+
+    public function test_add_taste()
```
Comment (nemoipaha) [inline tests/Feature/Api/UserTaste/TasteTest.php]: custom_taste

File: tests/Feature/Api/UserTaste/TasteTest.php [hunk 11]
```diff
@@ -37,4 +37,31 @@ public function test_get_tastes()
...
+                'name' => 'Custom taste'
+            ]
+        );
+        $response->assertStatus(201);
+        $this->assertDatabaseHas('tastes', [
+            'name' => 'Custom taste'
+        ]);
+    }
+
+    public function test_delete_taste()
```
Comment (nemoipaha) [inline tests/Feature/Api/UserTaste/TasteTest.php]: )

File: frontend/src/components/taste/TasteList.vue (hunk 1 above)
Comment (aLLeXUs) [reply frontend/src/components/taste/TasteList.vue]: Yes, it should be, api returns custom tastes only for current user, so we don`t need to check. But now we can use is_default to check taste is custom or not

**Pull request closed without merging**